core/src/components/item/item.md.scss

```diff
@@ -32,65 +32,22 @@
   text-transform: none;
 }
 
-:host(.item-fill-outline) {
-  --highlight-height: 2px;
-}
-
 // Item Fill: None
 // --------------------------------------------------
 
-:host(.item-fill-none.item-interactive.ion-focus) .item-highlight,
-:host(.item-fill-none.item-interactive.item-has-focus) .item-highlight,
-:host(.item-fill-none.item-interactive.ion-touched.ion-invalid) .item-highlight {
-  transform: scaleX(1);
-
-  border-width: 0 0 var(--full-highlight-height) 0;
-  border-style: var(--border-style);
-  border-color: var(--highlight-background);
-}
-
-:host(.item-fill-none.item-interactive.ion-focus) .item-native,
-:host(.item-fill-none.item-interactive.item-has-focus) .item-native,
-:host(.item-fill-none.item-interactive.ion-touched.ion-invalid) .item-native {
-  border-bottom-color: var(--highlight-background);
-}
-
-// Item Fill: Outline
-// --------------------------------------------------
-
-:host(.item-fill-outline.item-interactive.ion-focus) .item-highlight,
-:host(.item-fill-outline.item-interactive.item-has-focus) .item-highlight {
-  transform: scaleX(1);
-}
-
-:host(.item-fill-outline.item-interactive.ion-focus) .item-highlight,
-:host(.item-fill-outline.item-interactive.item-has-focus) .item-highlight,
-:host(.item-fill-outline.item-interactive.ion-touched.ion-invalid) .item-highlight {
-  border-width: var(--full-highlight-height);
-  border-style: var(--border-style);
-  border-color: var(--highlight-background);
-}
-
-:host(.item-fill-outline.item-interactive.ion-touched.ion-invalid) .item-native {
-  border-color: var(--highlight-background);
-}
-
-// Item Fill: Solid
-// --------------------------------------------------
-
-:host(.item-fill-solid.item-interactive.ion-focus) .item-highlight,
-:host(.item-fill-solid.item-interactive.item-has-focus) .item-highlight,
-:host(.item-fill-solid.item-interactive.ion-touched.ion-invalid) .item-highlight {
+:host(.item-interactive.ion-focus) .item-highlight,
+:host(.item-interactive.item-has-focus) .item-highlight,
+:host(.item-interactive.ion-touched.ion-invalid) .item-highlight {
   transform: scaleX(1);
 
   border-width: 0 0 var(--full-highlight-height) 0;
   border-style: var(--border-style);
   border-color: var(--highlight-background);
 }
 
-:host(.item-fill-solid.item-interactive.ion-focus) .item-native,
-:host(.item-fill-solid.item-interactive.item-has-focus) .item-native,
-:host(.item-fill-solid.item-interactive.ion-touched.ion-invalid) .item-native {
+:host(.item-interactive.ion-focus) .item-native,
+:host(.item-interactive.item-has-focus) .item-native,
+:host(.item-interactive.ion-touched.ion-invalid) .item-native {
```

Comment:
Since these styles previously only applied on `fill="solid"`, wouldn't we want to remove them completely rather than have them apply in all cases?

Reply:
fixed in https://github.com/ionic-team/ionic-framework/pull/29102/commits/e65dbf6f9a06a001fae0031e44cfe05d063c6a13